rebasehelper/checkers/licensecheck_tool.py

```diff
@@ -0,0 +1,130 @@
+# -*- coding: utf-8 -*-
+#
+# This tool helps you to rebase package to the latest version
+# Copyright (C) 2013-2014 Red Hat, Inc.
+#
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# he Free Software Foundation; either version 2 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License along
+# with this program; if not, write to the Free Software Foundation, Inc.,
+# 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
+#
+# Authors: Petr Hracek <phracek@redhat.com>
+#          Tomas Hozza <thozza@redhat.com>
+
+import os
+
+import six
+
+from rebasehelper.utils import ProcessHelper
+from rebasehelper.checker import BaseChecker
+
+
+class LicenseCheckTool(BaseChecker):
+    """license compare tool"""
+
+    CMD = "licensecheck"
+    DEFAULT = True
+    category = "SOURCE"
+
+    @classmethod
+    def match(cls, cmd=None):
+        if cmd == cls.CMD:
+            return True
+        else:
+            return False
+
+    @classmethod
+    def get_checker_name(cls):
+        return cls.CMD
+
+    @classmethod
+    def is_default(cls):
+        return cls.DEFAULT
+
+    @classmethod
+    def get_license_changes(cls):
```

Comment:
This is how I think the output of this method should look like:

```python
{
    'added': {
        'FDL': [
            'some/file.txt',
        ],
        'GPL': [
            'some/other/file.txt',
        ],
    },
    'changed': {
        'BSD => MIT': [
            'some/file.txt',
            'some/other/file.txt',
        ],
    },
    'removed': {
        'GPL': [
            'some/file.txt',
        ],
    },
}
```

where `added` are newly appearing files or transitions from `UNKNOWN`, and `removed` disappearing files or transitions to `UNKNOWN`.

Reply:
And the corresponding text output would be:
```
...
licensecheck
======
License changes occured!
Removed - license removed to unset or file disappeared
Added - license added(previously unset) or new file appeared

removed license(s)
====
* LGPL (v2 or later)
 - platforminputcontext/keyserver_x11.h
* GPL (v2 or later)
 - platforminputcontext/qfcitxplatforminputcontext.cpp
 - platforminputcontext/main.cpp
 - platforminputcontext/qfcitxplatforminputcontext.h
 - platforminputcontext/main.h
 - platforminputcontext/keydata.h
 - platforminputcontext/keyuni.h
 - platforminputcontext/keyuni.cpp
 - platforminputcontext/utils.h
 - platforminputcontext/utils.cpp

added license(s)
====
* BSD (3 clause)
 - COPYING.BSD
* UNKNOWN
 - quickphrase-editor/editor.ui
 - po/ko.po
...
```

Reply:
Well, the details should go to a licensecheck report file, only some summary (so maybe just a list of (dis)appearing licenses) should be printed in the general report.

Reply:
I am sorry, I meant the `report.txt` text output.

Reply:
Yes, that's what I mean by general report. `licensecheck` should have its own report file, just like e.g. `rpmdiff`.

Reply:
You actually got me right. Then I would suggest the `changed` licenses not the disappeared onces..?

Reply:
From the files point of view you have added, changed and removed licenses. But if you forget about files, and just take all the licenses present in the old sources and the new sources, then you can print only differences between those sets.